app/models/course.rb

```diff
@@ -98,4 +110,23 @@ def export_students_to_csv
       end
     end
   end
+
+  def delete_roster_students(spreadsheet, header_row_exists, perm_index)
+    perms = sort_perms_into_hash(spreadsheet, header_row_exists, perm_index)
+    self.roster_students.each do |student|
+      perm = perms[student.perm]
+      if perm.blank?
+        student.destroy
+      end
+    end
+  end
```

Comment:
I wonder if this function name is misleading / why this got its own function.

This appears to be filtering a roster for invalid students? Or perhaps deduplicating.

Would a better name be 'deduplicate_roster_students'

Or sanitize_roster_students